app/uk/gov/hmrc/tradergoodsprofiles/connectors/BaseConnector.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofiles.connectors
+
+import io.lemonlabs.uri.UrlPath
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.client.RequestBuilder
+import uk.gov.hmrc.tradergoodsprofiles.config.Constants
+
+trait BaseConnector {
+
+  val routerBaseRoute: String = "/trader-goods-profiles-router"
+
+  def routerRoute(eoriNumber: String, recordId: String): UrlPath =
+    UrlPath.parse(
+      s"$routerBaseRoute/$eoriNumber/records/$recordId"
+    )
+
+  implicit class HttpResponseHelpers(requestBuilder: RequestBuilder) {
+    def withClientId(implicit hc: HeaderCarrier): RequestBuilder =
```

Comment:
Here the client id would already be present in the header carrier and you're setting it again?

Reply:
HI Liandra, I see you point but we have tested that and it seems that we need to set the clientID explicitly, both integration test and postman tests fail if we do not set the clientID. I have see other repo to actually set this explicitly. What are your thought on this?

Reply:
Well if you look at the definition of setHeader - likely that your tests are wrong?
`Adds the header. If the header has already been defined (e.g. from HeaderCarrier), it will be replaced. It does not affect headers not mentioned.`